pmlc/ast/ast_ops.cc

```diff
@@ -133,13 +134,47 @@ struct GatherOp : Intrinsic {
           "to be a constant float");
     }
 
-    TensorShape shape{tensor.elementType};
-    for (auto i = 0; i < axis.getValue(); i++) {
-      shape.sizes.push_back(tensor.sizes[i]);
+    auto mode = getIntegerValue(evaluator, operands[6]);
+    if (!mode) {
+      throw std::runtime_error("'gather' primitive expects the 'mode' argument "
+                               "to be a constant integer");
+    }
```

Comment:
Please check idxs data type according to mode, since ND mode does not support float index.

Reply:
Checked.